### ISSUE TITLE
Introspection cache not working for me

### ISSUE THREAD

**Issue by harikant-lftt** [nokia/kong-oidc#219]
Hi 

I was trying to validate introspection cache, I have Kong OIDC plugin with Keycloak.

To test this scenario I have authenticated a user and made some api call, after that I stoped my Keycloak server for sometime and Kong started giving me 401 response.  It look like plugin is calling introspection endpoint each time and not storing  Authenticated token in cache.

Is there any special configuration option? which may be I missed to enabled it.

Thanks in advance for your help.

**Comment by gustoliv** [comment]
Hi @harikant-lftt, in my case to make introspection cache working I had to set the `lua_shared_dict` directive inside Kong Gateway.

There is a series of lua dicts used to make cache working, they are:

- discovery (cache OpenID configuration)
- jwks (cache JWKS)
- introspection (cache introspection responses)
- jwt_verification (cache JWT verification statuses)

This can be done via an environment variable named `KONG_NGINX_HTTP_LUA_SHARED_DICT` but this variable only allow to specify one dict at a time.

There is a hacky workaround to pass more than one dict placing `lua_shared_dict` directive before each dict delimited by a semicolon:

`KONG_NGINX_HTTP_LUA_SHARED_DICT=discovery 1m; lua_shared_dict jwks 1m; lua_shared_dict introspection 10m; lua_shared_dict jwt_verification 1m`

Pay attention to the last parameter passed, which CANNOT be ended with a semicolon, otherwise Kong will fail to start.

**Comment by harikant-lftt** [comment]
Hi @gustoliv 
<img width="497" alt="Image" src="https://github.com/user-attachments/assets/6c6607a4-1c8a-4a9e-b5ec-36d389b7f663" />

I am running in k8s, and I add this env variable, but that didn't help

- name: KONG_NGINX_HTTP_LUA_SHARED_DICT
          value: lua_shared_dict introspection 10m


Error: could not prepare Kong prefix at /usr/local/kong: nginx configuration is invalid (exit code 1):
nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /usr/local/kong/nginx.conf:6
nginx: [emerg] invalid number of arguments in "lua_shared_dict" directive in /usr/local/kong/nginx-kong.conf:32
nginx: configuration file /usr/local/kong/nginx.conf test failed


  Run with --v (verbose) or --vv (debug) for more details

**Comment by harikant-lftt** [comment]
- name: KONG_NGINX_HTTP_LUA_SHARED_DICT
   value: introspection 10m

this setup, help to solve the problem

**Comment by gustoliv** [comment]
> * name: KONG_NGINX_HTTP_LUA_SHARED_DICT
>   value: introspection 10m
> 
> this setup, help to solve the problem

Yes, the **first** lua dict must not be prefixed with "lua_shared_dict" otherwise the parse of the configuration will fail.
